python/lsst/afw/table/base/baseContinued.py

```diff
@@ -283,6 +283,24 @@ def asAstropy(self, cls=None, copy=False, unviewable="copy"):
             )
         return cls(columns, meta=meta, copy=False)
 
+    def __dir__(self):
+        # This custom dir is necessary due to the custom getattr below.
+        # Without it, not all of the methods available are returned with dir.
+        # See DM-7199
+        def recursive_get_class_dir(cls):
+            '''
+            Return a set containing the names of all methods
+            for a given class *and* all of its subclasses.
+            '''
+            result = set()
+            if cls.__bases__:
+                for subcls in cls.__bases__:
+                    result |= recursive_get_class_dir(subcls)
+            result |= set(cls.__dict__.keys())
+            return result
+        return sorted(set(dir(self.columns)) | set(dir(self.table)) |
```

Comment:
I didn't know about the set operators (pipe for union, in this case). That's cool.

Separately, I think this return should be separated from the above by a line, since it's not part of that inner function.